hazelcast/src/main/java/com/hazelcast/spi/discovery/impl/DiscoveryServicePropertiesBuilder.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.spi.discovery.impl;
+
+import com.hazelcast.config.InvalidConfigurationException;
+import com.hazelcast.config.properties.PropertyDefinition;
+import com.hazelcast.config.properties.ValidationException;
+import com.hazelcast.config.properties.ValueValidator;
+import com.hazelcast.core.TypeConverter;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import static com.hazelcast.util.MapUtil.createHashMap;
+
+/**
+ * Validates, verifies, and maps properties with the given definitions.
+ */
+class DiscoveryServicePropertiesBuilder {
+
+    /**
+     * Validates, verifies, and maps {@code properties} with the given {@code propertyDefinitions}.
+     *
+     * @param properties          properties from the Hazelcast node configuration (from the service-discovery section)
+     * @param propertyDefinitions property definitions specific for the given
+     * {@link com.hazelcast.spi.discovery.DiscoveryStrategy}
+     * @return mapped properties
+     * @throws InvalidConfigurationException if the the required properties are not satisfied or any property is not not
+     * applicable to the given definitions
+     * @throws ValidationException           if any property is invalid
+     */
+    Map<String, Comparable> buildProperties(Map<String, Comparable> properties,
```

Comment:
Since this method doesn't require any state from the DiscoveryServicePropertiesBuilder, the method can be static. Also the name Builder is confusing since it isn't builder as we are using in the rest of the system like SerializationServiceBuilder. This is more of a utility method.

Reply:
Completely agree with the builder naming. I've changed it. Concerning the static utility, I've also changed it. Here, my reasoning behind the non-static class was mainly that the DefaultDiscoveryService could be later refactored to more classes which are injected and therefore better testable. But, we can leave it for later.

I've created a separate class for two reasons: 
- testability
- DefaultDiscoveryService is already way too big, it definitely covers more than one responsibility. Mapping properties is a separate responsibility than creating the discovery strategy.

Do you have any reasons it could better off as a part of DefaultDiscoveryService?